server/views/pages/recategoriserHome.html

```diff
@@ -95,24 +100,24 @@ <h1 class="govuk-heading-l">{{ pageTitle }}</h1>
           {% endset -%}
 
           {% set filterCategories = [] %}
-          {% if filters.suitabilityForOpenConditions.length > 0 %}
+          {% for key, filterNames in filters %}
             {% set selectedItems = [] %}
-            {% for filter in filters.suitabilityForOpenConditions %}
-              {% set startPositionOfFilterInUrl = fullUrl.indexOf(filter) - (28 + 7) %}
-              {% set removalHref = fullUrl.substring(0, (startPositionOfFilterInUrl - 1) if fullUrl[startPositionOfFilterInUrl - 1] == '&' or filters.suitabilityForOpenConditions.length == 1 else startPositionOfFilterInUrl) + fullUrl.substr(fullUrl.indexOf(filter) + filter.length + (0 if fullUrl[startPositionOfFilterInUrl - 1] == '&' else 1), fullUrl.length) %}
+            {% for filter in filterNames %}
+              {% set startPositionOfFilterInUrl = fullUrl.indexOf(filter) - (key.length + 7) %}
+              {% set removalHref = fullUrl.substring(0, (startPositionOfFilterInUrl - 1) if fullUrl[startPositionOfFilterInUrl - 1] == '&' or numberOfFiltersApplied == 1 else startPositionOfFilterInUrl) + fullUrl.substr(fullUrl.indexOf(filter) + filter.length + (0 if fullUrl[startPositionOfFilterInUrl - 1] == '&' else 1), fullUrl.length) %}
```

Comment:
I have stared at this for a while and don't understand what it's trying to do

it _feels_ like you have worked magic to do whatever this does, and I appreciate the hard work put in

however, I imagine this could be done in a more maintainable manner by running the `fullUrl` through a custom filter - https://mozilla.github.io/nunjucks/api#custom-filters - and then doing the heavy lifting in javascript?